lib/bootboot/ruby_source.rb

```diff
@@ -18,15 +18,30 @@ def ruby_spec_name
 
     def specs
       Bundler::Index.build do |idx|
-        # If the ruby version specified in the Gemfile is different from the
-        # Ruby version currently running, we want to build a definition without
-        # a lockfile (so that `ruby_version` in the Gemfile isn't overridden by
-        # the lockfile) and get its `ruby_version`. This will be used both
-        # during dependency resolution so that we can pretend the intended Ruby
-        # version is present, as well as when updating the lockfile itself.
-        ruby_version = Bundler::Definition.build(Bootboot::GEMFILE, nil, false).ruby_version
-        ruby_version ||= Bundler::RubyVersion.system
-        ruby_spec = Gem::Specification.new(ruby_spec_name, ruby_version.gem_version)
+        requested_ruby = Bundler::Definition.build(Bootboot::GEMFILE, nil, false).ruby_version
+        system_version = Bundler::RubyVersion.system.gem_version
+        requirement = Gem::Requirement.new(requested_ruby.versions) unless requested_ruby.nil?
```

Comment:
nit:
```suggestion
        requirement = Gem::Requirement.new(requested_ruby.versions) if requested_ruby
```

Sorry I might have used `.nil?` in my own suggestion but generally I prefer to rely on value being thuthy/falsey